File: app/src/main/java/com/demo/AppV1.java
package com.demo;

import com.demo.cg.IntegerAdder;
import net.sf.cglib.proxy.Enhancer;
import net.sf.cglib.proxy.NoOp;

public class AppV1 {

    public static void main(String[] args) {
        NoOp noOp = NoOp.INSTANCE;

        IntegerAdder calculator =
                (IntegerAdder) Enhancer.create(IntegerAdder.class, noOp);

        int result = calculator.add(1, 2);
        System.out.println("result: " + result);
    }
}

File: app/src/main/kotlin/com/demo/AppV1.kt
package com.demo

import net.sf.cglib.proxy.Enhancer
import net.sf.cglib.proxy.NoOp
import com.demo.cg.IntegerAdder

object AppV1 {
    @JvmStatic
    fun main(args: Array<String>) {
        val noOp = NoOp.INSTANCE
        val calculator =
            Enhancer.create(IntegerAdder::class.java, noOp) as IntegerAdder
        val result = calculator.add(1, 2)
        println("result: $result")
    }
}